src/Analyzers/Core/CodeFixes/RemoveUnusedParametersAndValues/AbstractRemoveUnusedValuesCodeFixProvider.cs

```diff
@@ -731,7 +736,7 @@ private async Task<SyntaxNode> ReplaceDiscardDeclarationsWithAssignmentsAsync(Sy
             }
 
             var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-            return await service.ReplaceAsync(memberDeclaration, semanticModel, cancellationToken).ConfigureAwait(false);
+            return await service.ReplaceAsync(memberDeclaration, semanticModel, document.Project.Solution.Workspace, cancellationToken).ConfigureAwait(false);
```

Comment:
So the service creates a new SyntaxEditor using [this](http://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/Editing/SyntaxEditor.cs,36) internal API. I had to switch to the [public API](http://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/Editing/SyntaxEditor.cs,24) that takes `Workspace` so it can be moved to the shared layer.

https://github.com/dotnet/roslyn/pull/42397/files#diff-9b36fbb086de7b1457f140d61ecd1832R34